### ISSUE TITLE
Domain name not resolving correctly on custom rules.

### ISSUE THREAD

**Issue by gNash-dev** [henrypp/simplewall#1012]
When I create a rule with a domain name instead of an IP, initially it gets resolved and the rule works as expected. However when the IP that the domain was pointing to changes, it does not get resolved again.

**Comment by lumpov** [comment]
Confirm. Rules based on domains, which uses Cloudflare DNS with DDOS protect or hide real IP doesn't work properly. For example, api.binance.com which ip changes periodically. Ping this domain show me current IP and I add it to rule, but this is wrong case.

**Comment by henrypp** [comment]
wfp is not designed to be dns resolver.

**Comment by lumpov** [comment]
> wfp is not designed to be dns resolver.

maybe it is possible that the WFP will be able to use correct DNS resolve of domains in rules?

dns resolver must be the specific windows service, but wfp use it to resolve  use domains in rules is more comfortable than ip.

**Comment by henrypp** [comment]
> maybe it is possible that the WFP will be able to use correct DNS resolve of domains in rules?

or maybe use dns resolver to resolve dns?

**Comment by lumpov** [comment]
For sure, I don't know how to great app Simplewall and WFP works when I use FQDN in rule. But FQDN in rule works not well in cases above - this is feedback only. If it possible to fix this problem will be great!

**Comment by lumpov** [comment]
Every day I need to check ip of `api.binance.com` (via ping) and if it is new I write it in rule, because `api.binance.com` in rule don't work (other FQDN with static DNS working well).

Why for ping-command Windows correct resolve FQDN, but in the same time, on the same PC WFP use old resolved ip of `api.binance.com`?

I try to refresh rules, disable/enable rules with no luck.

I hope this problem can be solved.

**Comment by gNash-dev** [comment]
I agree with @lumpov, such a feature would help a ton, when having to deal with dynamic ips.

**Comment by henrypp** [comment]
Block FQDN names in WFP is highly unsafe, because of WFP is filtering only by ip addresses (not hostname), so blocking FQDN throught WFP be like that:

> but in the same time, on the same PC WFP use old resolved ip

Standart system dnscache resolver is not good for this idea too, because of `hosts` file can be skipped via `DNS_QUERY_NO_HOSTS_FILE`.

So i will suggest you to start using lightweight and well configurable [dnscrypt-proxy](https://github.com/DNSCrypt/dnscrypt-proxy). It can be configured to block every FQDN and domain masks you wanted.

**Comment by lumpov** [comment]
thank you for not stay indifferent!

Sorry, but I not understand, how `dnscrypt-proxy` can help me in my case? Enable all connections through `simplewall` and filter everything with FQDN via `dnscrypt-proxy`?

May be try to find right way to resolve FQDN in WFP? Or other solution.

**Comment by henrypp** [comment]
> Sorry, but I not understand, how dnscrypt-proxy can help me in my case?

No one can access blocked domain in any way, even with `DNS_QUERY_NO_HOSTS_FILE` flag.

> May be try to find right way to resolve FQDN in WFP?

No. Even static domain ip's can change ip on fly, and simplewall is not that tool who responsible to domain name resolution.

I mean, this out of logic, and only one solution for this is using correct tool to block domain names.

**Comment by lumpov** [comment]
Sorry, but I explain in Russian.

Уважаемый Henry!

Не могу понять, если я задал правило WFP через simplewall с указанием FQDN и соответствующее правилу приложение делает запрос к указанному (в правиле) FQDN, то почему оно (правило) при смене IP перестаёт работать, т.е. перестаёт разрешать сетевой трафик по FQDN?

Ведь в правиле указан FQDN, приложение запрашивает по FQDN. Да, IP изменился, но ни в правилах не задается IP, ни приложение не обращается по IP.

Да, я понимаю, что simplewall не ответственный за резолвинг, ответственная операционная система, а на примере ping домен резолвится верно, значит не в этом дело, что-то работает не так, я не знаю, может быть WFP резолвит из кэша FQDN, может быть можно его обновлять кэш или не использовать совсем. Или simplewall задает в правило IP домена на момент сохранения правила и соответственно правило работает по IP?

Может быть я натолкну вас на идею?
